eam-light-backendweb/src/main/java/ch/cern/cmms/eamlightweb/meter/MeterRest.java

```diff
@@ -83,18 +85,35 @@ public Response createReading(MeterReading meterReading) {
 	}
 
 	private List<MeterReadingWrap> getMeterEquipmentMeterReading(String equipmentCode, String meterCode) throws InforException {
-		GridRequest gridRequest = new GridRequest("OSEQMR", GridRequest.GRIDTYPE.LIST);
+		GridRequest gridRequest = new GridRequest("OSMETE", GridRequest.GRIDTYPE.LIST);
+		gridRequest.setUserFunctionName("OSOBJA");
+
+		gridRequest.addParam("parameter.organization", authenticationTools.getR5InforContext().getOrganizationCode());
 		gridRequest.addParam("parameter.object", equipmentCode);
 
 		if (meterCode != null) {
-			gridRequest.addFilter("physicalmeter", meterCode, "=");
+			gridRequest.addFilter("metercode", meterCode, "=");
 		}
 
+		GridRequestResult gridResult = inforClient.getGridsService().executeQuery(authenticationTools.getR5InforContext(), gridRequest);
+
 		List<MeterReadingWrap> result = convertGridResultToObject(MeterReadingWrap.class,
-				null,
-				inforClient.getGridsService().executeQuery(authenticationTools.getInforContext(), gridRequest));
+				null, gridResult
+		);
 
-		result.forEach(meter -> meter.setEquipmentCode(equipmentCode));
+		result.forEach(meter -> {
+			try {
+			meter.setEquipmentCode(equipmentCode);
+				GridRequest uomGridRequest = new GridRequest("BSUOMS", GridRequest.GRIDTYPE.LIST);
+				uomGridRequest.addFilter("uomcode", meter.getUom(), "=");

```

Comment:
Change so the requests are not done sequentially, you can create the filter with all the required uoms, and then a map with the descriptions, and then get assign them at once